### PR TITLE
Adds PIVOT and SCAN_INDEXED to evaluation engine

### PR DESCRIPTION
**Pull request by RCHowell**
## Description

- Fixes scan_indexed in the PlanTyper
- Adds PIVOT expression to engine
- Adds SCAN_INDEXED relation operator to engine

## Other Information
- Updated Unreleased Section in CHANGELOG: **[YES/NO]**
No

- Any backward-incompatible changes? **[YES/NO]**
No

- Any new external dependencies? **[YES/NO]**
No

- Do your changes comply with the [Contributing Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CONTRIBUTING.md)
  and [Code Style Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CODE_STYLE.md)? **[YES/NO]**
Yes

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/PlanTyper.kt [hunk 1]
```diff
@@ -93,6 +95,7 @@ import org.partiql.value.BoolValue
 import org.partiql.value.PartiQLValueExperimental
 import org.partiql.value.TextValue
 import org.partiql.value.boolValue
+import java.sql.Struct
```
Comment (RCHowell) [inline partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/PlanTyper.kt:98]: ????

**Comment by RCHowell** [comment]
Waiting to rebase on https://github.com/partiql/partiql-lang-kotlin/pull/1297

File: partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rel/RelScanIndexed.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    override fun next(): Record? {
+        if (!iterator.hasNext()) {
+            return null
+        }
+        val i = BigInteger.valueOf(index)
+        val v = iterator.next()
+        index += 1
+        return Record.of(v, intValue(i))
```
Comment (johnedquinn) [inline partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rel/RelScanIndexed.kt:35]: From what we discussed, I thought we would be returning an in64Value

File: partiql-eval/src/test/kotlin/org/partiql/eval/internal/PartiQLEngineDefaultTest.kt [hunk 3]
```diff
@@ -66,4 +69,49 @@ class PartiQLEngineDefaultTest {
...
+           >> AS x
+            """.trimIndent()
+        ).root
+        val session = PartiQLPlanner.Session("q", "u")
+        val plan = planner.plan(statement, session)
+
+        val prepared = engine.prepare(plan.plan)
+        val result = engine.execute(prepared) as PartiQLResult.Value
+        val output = result.value as BagValue<*>
+        println(output.toIon().toString())
```
Comment (johnedquinn) [inline partiql-eval/src/test/kotlin/org/partiql/eval/internal/PartiQLEngineDefaultTest.kt:115]: #1294 is merged. You can pull/assert

**Review by johnedquinn — CHANGES_REQUESTED** [review]

File: partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rel/RelScanIndexed.kt (hunk 2 above)
Comment (RCHowell) [reply partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rel/RelScanIndexed.kt:35]: Yes, had to merge that typing udpate to main first

**Comment by RCHowell** [comment]
Will rebase on https://github.com/partiql/partiql-lang-kotlin/pull/1305

**Comment by RCHowell** [comment]
Replaced by https://github.com/partiql/partiql-lang-kotlin/pull/1316

**Pull request closed without merging**